hie-plugin-api/Haskell/Ide/Engine/PluginsIdeMonads.hs

```diff
@@ -131,163 +148,56 @@ instance ToJSON IdePlugins where
 
 type IdeGhcM = GM.GhcModT IdeM
 
-instance MonadMTState IdeState IdeGhcM where
-  readMTS = lift $ lift $ lift readMTS
-  modifyMTS f = lift $ lift $ lift $ modifyMTS f
+newtype IDErring m a = IDErring { getIDErring :: ExceptT IdeError m a } deriving
+  (Functor, Applicative, Monad, MonadReader r, MonadState s, MonadIO, MonadTrans, MonadBase b, MFunctor)
+instance GM.MonadIO m => GM.MonadIO (IDErring m) where liftIO = lift . GM.liftIO
+instance GM.GmEnv m => GM.GmEnv (IDErring m) where gmeAsk = lift GM.gmeAsk; gmeLocal f x = liftWith (\run -> GM.gmeLocal f $ run x) >>= restoreT . return
+instance GM.GmLog m => GM.GmLog (IDErring m) where gmlJournal = lift . GM.gmlJournal; gmlHistory = lift GM.gmlHistory; gmlClear = lift GM.gmlClear
+instance GM.GmOut m => GM.GmOut (IDErring m) where gmoAsk = lift GM.gmoAsk
+instance GM.GmState m => GM.GmState (IDErring m) where gmsGet = lift GM.gmsGet; gmsPut = lift . GM.gmsPut; gmsState = lift . GM.gmsState
+instance (Functor f, MonadFree f m) => MonadFree f (IDErring m) where wrap x = liftWith (\run -> wrap $ fmap run x) >>= restoreT . return
+
```

Comment:
Please do not let lines get longer than 115 chars (preferably 80), else the github UI does not show them properly

Reply:
Huh, shows fine for me. I liked hiding the obvious boilerplate way to the right.